source/common/filter/http/filter_config_discovery_impl.cc

```diff
@@ -150,12 +179,22 @@ void FilterConfigSubscription::onConfigUpdate(
     const std::vector<Config::DecodedResourceRef>& added_resources,
     const Protobuf::RepeatedPtrField<std::string>& removed_resources, const std::string&) {
   if (!removed_resources.empty()) {
-    ENVOY_LOG(error,
-              "Server sent a delta ExtensionConfigDS update attempting to remove a resource (name: "
-              "{}). Ignoring.",
-              removed_resources[0]);
-  }
-  if (!added_resources.empty()) {
+    ASSERT(removed_resources.size() == 1);
+    ENVOY_LOG(debug, "Removing filter config {}", filter_config_name_);
+    const auto pending_update = std::make_shared<std::atomic<uint64_t>>(
```

Comment:
This "wait for all threads to execute and then callback" pattern seems likely to be used in many places. We already do something like this for post-TLS completion across threads. Should there be some more standard utility for this in `source/common/common`? Seems bad to have to manually compute the number of threads here..

Reply:
Looking around I couldn't find something that matches it exactly, and I'm not sure if it's completely correct: would other threads (e.g. access logger thread) also be included in `runOnAllThreads`?

I think the better approach here would be to piggy back off the post-TLS completion callback, that way I believe we don't need to care about the number of threads at all. Let me look into this real quick